File: src/main/java/pages/ProductDetailPage.java
package pages;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class ProductDetailPage extends BasePage{
    @FindBy(css = "h2.name")
    WebElement model;
    @FindBy(css = "h3.price-container")
    WebElement price;
    @FindBy(linkText = "Add to cart")
    WebElement addToCartButton;

    public ProductDetailPage() {
        this.driver = getDriver();
        PageFactory.initElements(driver, this);
    }

    public String getModel() {
        return getTextFromWebElement(model);
    };
    public String getPrice() {
        String priceText = getTextFromWebElement(price);
        int inicioPrice = priceText.indexOf("$") + 1;
        int finPrice = priceText.indexOf("*") - 1;
        return (priceText.substring(inicioPrice, finPrice));
    };
    public void clickAddToCartButton() {
        clickOnWebElement(addToCartButton);
    };
}

File: Desktop/proyectoWdio/test/pageobjects/productDetailPage.js
const Page = require('./page');

/**
 * sub page containing specific selectors and methods for a specific page
 */
class ProductsDetailPage extends Page {
    /**
     * define selectors using getter methods
     */
    get model(){
        return $("//h2[@class='name']")
    };
    get price(){
        return $("//h3[@class='price-container']")
    };
    get addToCartButton(){
        return $("//a[text()='Add to cart']")
    };

    async getModel() {
        await this.getTextFromWebElement(this.model);
    }
    async getPrice() {
        const priceText = await this.getTextFromWebElement(this.price);
        const inicioPrice = priceText.indexOf("$") + 1;
        const finPrice = priceText.indexOf("*") - 1;
        return (priceText.substring(inicioPrice, finPrice));
    }

    async clickAddToCart() {
        await this.clickOnWebElement(this.addToCartButton);
    }
}

module.exports = new ProductsDetailPage();